src/vm_cpu.rs

```diff
@@ -777,6 +969,75 @@ impl VmCpu {
 
         Ok(ActiveVmCpu::restore_from(self, vm_pages, parent_vcpu))
     }
+
+    /// Updates internal state for PMU counters.
+    /// This should be called following a successful SBI call to configure counters.
+    pub fn update_configured_counters(
+        &mut self,
+        counter_index: u64,
+        config_flags: PmuCounterConfigFlags,
+        event_type: PmuEventType,
+        event_data: u64,
+    ) {
+        self.pmu_state.update_configured_counters(
```

Comment:
optional: instead of having all these wrappers you could expose a reference directly to `self.pmu_state`.